src/java/fr/paris/lutece/plugins/forms/validation/AbstractDateValidator.java

```diff
@@ -72,11 +72,14 @@ public boolean validate( FormQuestionResponse questionResponse, Control control
             Date dateControl = new Date( dateFormat.parse( control.getValue( ) ).getTime( ) );
             for ( Response response : questionResponse.getEntryResponse( ) )
             {
-                Date date = new Date( Long.parseLong( response.getResponseValue( ) ) );
-                if ( !validateDate( dateControl, date ) )
-                {
-                    return false;
-                }
+            	if (response.getResponseValue() != null)
+            	{
+            		Date date = new Date( Long.parseLong( response.getResponseValue( ) ) );
+                    if ( !validateDate( dateControl, date ) )
+                    {
+                        return false;
+                    }
+            	}
             }
             return true;
```

Comment:
if the date should be before or after a given date, but the response is null, we maybe should not return true ?

Reply:
I think it's that way because it follow the app logic. There is 2 seperated controls, the "mandatory" option in "Modify the question", and the date controls in "Manage validation".
If the question has not "mandatory" checked it mean we can give a null value.
It would be strange to have a question marked as "not mandatory" with a validation error if no values.

Reply:
This test does not make sense for a null date.
The date control should remain "context independent" and not return TRUE if the date is null.
We should test upstream  if the value is null, (for example in FormResponseManager.validateFormResponses() , and in that case not apply checks (dates or others).
Mandatory control will be performed independently.

Reply:
I propose the solution below, which is less complex: we return false by default, or the result of the validation test only if we can execute it.
On the other hand, if there are several entries for the question, the code will have to be adapted.


```
   public boolean validate( FormQuestionResponse questionResponse, Control control )
    {
        SimpleDateFormat dateFormat = new SimpleDateFormat( "yyyy-MM-dd" );
        try
        {
            Date dateControl = new Date( dateFormat.parse( control.getValue( ) ).getTime( ) );
            for ( Response response : questionResponse.getEntryResponse( ) )
            {
            	if (response.getResponseValue() != null)
            	{
            	     Date date = new Date( Long.parseLong( response.getResponseValue( ) ) );
                    return validateDate( dateControl, date ) );
                }
            }
        }  
        catch( ParseException e )
        {
            AppLogService.error( "Error formatingdate", e );
        }
        
        return false;
        
    }
```